lib-impl/src/main/java/org/ergoplatform/appkit/impl/NodeAndExplorerDataSource.java

```diff
@@ -0,0 +1,195 @@
+package org.ergoplatform.appkit.impl;
+
+import com.google.common.base.Preconditions;
+
+import org.ergoplatform.ErgoLikeTransaction;
+import org.ergoplatform.appkit.Address;
+import org.ergoplatform.appkit.BlockchainDataSource;
+import org.ergoplatform.appkit.ErgoClient;
+import org.ergoplatform.appkit.ErgoClientException;
+import org.ergoplatform.appkit.InputBox;
+import org.ergoplatform.appkit.Iso;
+import org.ergoplatform.appkit.SignedTransaction;
+import org.ergoplatform.appkit.BlockHeader;
+import org.ergoplatform.appkit.BlockchainParameters;
+import org.ergoplatform.explorer.client.DefaultApi;
+import org.ergoplatform.explorer.client.ExplorerApiClient;
+import org.ergoplatform.explorer.client.model.OutputInfo;
+import org.ergoplatform.restapi.client.ApiClient;
+import org.ergoplatform.restapi.client.BlocksApi;
+import org.ergoplatform.restapi.client.ErgoTransaction;
+import org.ergoplatform.restapi.client.ErgoTransactionDataInput;
+import org.ergoplatform.restapi.client.ErgoTransactionInput;
+import org.ergoplatform.restapi.client.ErgoTransactionOutput;
+import org.ergoplatform.restapi.client.InfoApi;
+import org.ergoplatform.restapi.client.NodeInfo;
+import org.ergoplatform.restapi.client.TransactionsApi;
+import org.ergoplatform.restapi.client.UtxoApi;
+import org.ergoplatform.restapi.client.WalletApi;
+import org.ergoplatform.restapi.client.WalletBox;
+
+import java.math.BigDecimal;
+import java.util.ArrayList;
+import java.util.List;
+
+import jline.internal.Nullable;
+import okhttp3.OkHttpClient;
+import retrofit2.Call;
+import retrofit2.Response;
+import retrofit2.Retrofit;
+import special.sigma.Header;
+
+public class NodeAndExplorerDataSource implements BlockchainDataSource {
```

Comment:
Please describe how this implementation implements the interface. Also suggest to rename the class so it ends with Impl